### ISSUE TITLE
ASCII / unicode exception

### ISSUE THREAD

**Issue by ronaldknol** [shirosaidev/diskover#6]
I got this error on a run

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/ronald/diskover/diskover.py", line 260, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYS, MINSIZE, EXCLUDED_FILES, VERBOSE, DEBUG)
  File "/home/ronald/diskover/diskover.py", line 225, in crawlFiles
    filemeta = '{"filename": "%s", "extension": "%s", "path_full": "%s", "path_parent": "%s", "filesize": %s, "owner": "%s", "group": "%s", "last_modified": %s, "last_access": %s, "last_change": %s, "hardlinks": %s, "inode": %s, "indexing_date": %s}' % (name.decode('utf-8'), extension, filename_fullpath.decode('utf-8'), abspath.decode('utf-8'), size, owner, group, mtime, atime, ctime, hardlinks, inode, indextime)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 8: ordinal not in range(128)
```

I tweaked the code to add a print in a try/except case, and I see that these are the files causing the failures:

```
# ls -alb 02514* 00884*
-rw-rw-rw- 1 4294967294 root 26785 Oct  9  1991 02514\ St.\ Patrick’s\ Day
-rw-rw-rw- 1 4294967294 root 7043 Mar 13  1989 00884\ 5.25\ Disk
```

I have no idea which character is in the second filename, but I think it's just a quote in the first one.

Files in the filesystem are archives from old Mac (HFS) backups.
Back in the day the Mac operators would name their files anything they wanted... :)

**Comment by CopyRunStart** [comment]
I'm having the same issue.  This NAS is generally used by our OSX users so it may connect with what Ronald is saying.

When the run finished, it showed only 3.8TB instead of the actual 13TB.

`python /diskover/diskover/diskover.py -m 0 -s 0


    _/_/_/    _/            _/
   _/    _/        _/_/_/  _/  _/      _/_/    _/      _/    _/_/    _/  _/_/
  _/    _/  _/  _/_/      _/_/      _/    _/  _/      _/  _/_/_/_/  _/_/
 _/    _/  _/      _/_/  _/  _/    _/    _/    _/  _/    _/        _/
_/_/_/    _/  _/_/_/    _/    _/    _/_/        _/ v1.0.3   _/_/_/  _/
                              https://github.com/shirosaidev/diskover


[2017-05-01 14:38:43] [status] Connecting to Elasticsearch
[2017-05-01 14:38:43] [info] Checking for ES index
[2017-05-01 14:38:43] [warning] ES index exists, deleting
[2017-05-01 14:38:44] [info] Creating ES index
[2017-05-01 14:38:44] [status] Finding directories to crawl
Crawling: [21%] |████████--------------------------------| 6630/31279Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/diskover/diskover/diskover.py", line 263, in processDirectoryWorker
    indexAdd(ES, INDEXNAME, filelist, VERBOSE)
  File "/diskover/diskover/diskover.py", line 365, in indexAdd
    if DEBUG:
NameError: global name 'DEBUG' is not defined

Crawling: [34%] |██████████████--------------------------| 53597/156364Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/diskover/diskover/diskover.py", line 260, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYS, MINSIZE, EXCLUDED_FILES, VERBOSE, DEBUG)
  File "/diskover/diskover/diskover.py", line 225, in crawlFiles
    filemeta = '{"filename": "%s", "extension": "%s", "path_full": "%s", "path_parent": "%s", "filesize": %s, "owner": "%s", "group": "%s", "last_modified": %s, "last_access": %s, "last_change": %s, "hardlinks": %s, "inode": %s, "indexing_date": %s}' % (name.decode('utf-8'), extension, filename_fullpath.decode('utf-8'), abspath.decode('utf-8'), size, owner, group, mtime, atime, ctime, hardlinks, inode, indextime)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xef in position 2: ordinal not in range(128)


[2017-05-01 14:47:39] [info] Directories Crawled: 156364
[2017-05-01 14:47:39] [info] Files Indexed: 695656
[2017-05-01 14:47:39] [info] Elapsed time: 536.824841022
`

**Comment by shirosaidev** [comment]
I released version **1.0.4** tonight which should fix this issue. Please update and reply if it's fixed.

**Comment by CopyRunStart** [comment]
```
python /diskover/diskover/diskover.py -m 1 -s 1
[2017-05-04 00:23:15] [status] Connecting to Elasticsearch
[2017-05-04 00:23:15] [info] Checking for ES index: diskover-2017.04.22
[2017-05-04 00:23:15] [warning] ES index exists, deleting
[2017-05-04 00:23:15] [info] Creating ES index
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/diskover/diskover/diskover.py", line 314, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYSOLD, MINSIZE, EXCLUDED_FILES, VERBOSE)
  File "/diskover/diskover/diskover.py", line 270, in crawlFiles
    "owner": "%s" % owner.decode('utf-8'),
AttributeError: 'int' object has no attribute 'decode'

Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/diskover/diskover/diskover.py", line 314, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYSOLD, MINSIZE, EXCLUDED_FILES, VERBOSE)
  File "/diskover/diskover/diskover.py", line 270, in crawlFiles
    "owner": "%s" % owner.decode('utf-8'),
AttributeError: 'int' object has no attribute 'decode'

[2017-05-04 00:23:15] [status] Finding directories to crawl
[2017-05-04 00:27:19] [status] Finished crawling
[2017-05-04 00:27:19] [info] Checking for ES index: diskover_dupes-2017.04.22
[2017-05-04 00:27:19] [info] Creating ES index
[2017-05-04 00:27:19] [info] Directories Crawled: 156540
[2017-05-04 00:27:19] [info] Files Indexed: 0
[2017-05-04 00:27:19] [info] Elapsed time: 244.819179058

```

**Comment by shirosaidev** [comment]
I've released version 1.0.5 which should fix this. This is duplicate of issue #4 . Closing this issue.
